### PR TITLE
s22-4pm-team1-Harry Li  added section and page and navbar

### PR DESCRIPTION
**Pull request by haoyuanli2023**
# Overview

This PR adds the new section search index page so that users can get section results based on the inputted queries and have them displayed on a section table.

# Issues Addressed

#10 #28 
**Added** Sectionsearch to navbar and adding navbar test(passed) for sectionsearch.
**Added** page for sectionsearch and test for it (passed).
**Added** Storybook for SectionSearchesIndex page and successfully built. 
**Added** sectionsearch to App.js and test for it (passed). 
**Added** test for "convertToTimeRange" function in timeUtils and passed. 
**Added** test for "formatDays" with null case. 

# Storybook 

[Section Index Page](https://ucsb-cs156-s22.github.io/s22-4pm-courses-docs-qa/storybook-qa/Harry-Section-Navbar/?path=%2Fstory%2Fpages-sectionsearches-sectionsearchesindexpage--default)

[Nav Bar](https://ucsb-cs156-s22.github.io/s22-4pm-courses-docs-qa/storybook-qa/Harry-Section-Navbar/?path=%2Fstory%2Fcomponents-nav-appnavbar--basic-logged-in-admin-user)

# Details

## NEW Section Index Page

Sectionsearch Index page story

Sectionsearch

# Test Plan (for interactive testing) 

1. Run the backend and frontend on local machine
2. Proceed to section search index page via the navbar
3. Make a new search and verify that results appear on the section table

### CODE REVIEW

File: frontend/src/main/components/Nav/AppNavbar.js [hunk 1]
```diff
@@ -59,6 +59,16 @@ export default function AppNavbar({ currentUser, systemInfo, doLogout, currentUr
               }
             </Nav>
 
+            <Nav className="mr-auto">
+              {
+                hasRole(currentUser, "ROLE_USER") && (
+                  <NavDropdown title="Section Searches" id="appnavbar-section-searches-dropdown" data-testid="appnavbar-section-searches-dropdown" >
+                    <NavDropdown.Item href="/sectionsearches/search" data-testid="appnavbar-section-searches-search">search</NavDropdown.Item>
```
Comment (roberto-garcia-0) [inline frontend/src/main/components/Nav/AppNavbar.js:66]: Can you capitalize the `search` option?

File: frontend/src/main/components/Sections/SectionsTable.js [hunk 2]
```diff
@@ -45,6 +45,10 @@ export default function SectionsTable({ sections }) {
         {
             Header: 'Time',
             accessor: (row) => formatTime(row.section.timeLocations),
+            // accessor: (_row) => {
+            //     console.log("_row=",_row);
+            //     return "TBD";
+            // },
```
Comment (roberto-garcia-0) [inline frontend/src/main/components/Sections/SectionsTable.js:51]: Remove comment code.

**Review by roberto-garcia-0 — CHANGES_REQUESTED** [review]
Just need to remove some commented out code and fix a minor typo, and it's good to go!

**Review by roberto-garcia-0 — DISMISSED** [review]
LGTM!

**Review by pconrad — CHANGES_REQUESTED** [review]
One small .js that should be removed; otherwise LGTM!  Tested on QA

**Review by pconrad — APPROVED** [review]

**Review by SawyerEssabhoy — APPROVED** [review]
LGTM!

**Review by roberto-garcia-0 — APPROVED** [review]

**Pull request merged**